src/client/response.rs

```diff
@@ -41,22 +41,25 @@ impl Response {
     }
 
     pub async fn get(&mut self) -> Result<Option<ResponseBody>> {
-        let timeout = util::timestamp() - self.deadline;
+        let timeout = util::timestamp() > self.deadline;
 
-        if timeout == 0 || self.response_num == 0 {
+        if timeout || self.response_num == 0 {
             return Ok(None);
         }
-
-        let msg: Option<Message> = {
+        let msg: Message = {
```

Comment:
This should be `Option<Message>` because if the pop timedout it will return a None. A None is not a failure a None would means you waited to long that u exceeded the timeout hence there are no expected messages anymore.

There is a reason why I have changed it to Option<Message> from Message in the first place. Hence I was expecting you to ask why instead of just changing it back.